src/zarr_benchmarks/parse_json_for_plots.py

```diff
@@ -76,161 +79,334 @@ def get_benchmarks_dataframe(
+        col_wrap = None
+        plot_name = output_filename
+    else:
+        facet_kws = dict(sharex=True, sharey=True)
+        if len(data[col].unique()) < 3:
+            col_wrap = 2
+        else:
+            col_wrap = 3
+        plot_name = output_filename + "_subplots"
+
     graph = sns.relplot(
         data=data,
         x=x_axis,
         y=y_axis,
         hue=hue,
         style=hue,
         size=size,
+        col=col,
         height=4,
         aspect=1.5,
+        facet_kws=facet_kws,
+        col_wrap=col_wrap,
     )
-    x_axis_label, y_axis_label = get_axis_labels(x_axis, y_axis, group)
+    x_axis_label, y_axis_label = get_axis_labels(data, x_axis, y_axis)
     graph.set_axis_labels(x_axis_label, y_axis_label)
 
-    graph.figure.suptitle(title)
-    graph.figure.subplots_adjust(top=0.9)
-
-    if output_filename is not None:
-        save_plot_as_png(
-            graph,
-            Path(__file__).parents[2]
-            / "data"
-            / "plots"
-            / f"{group}_relplot_{output_filename}.png",
-        )
-
+    if title is not None:
+        graph.figure.suptitle(title)
+        graph.figure.subplots_adjust(top=0.9)
 
-def plot_relplot_subplots_benchmarks(
-    data: pd.DataFrame,
-    *,
-    group: str,
-    x_axis: str,
-    y_axis: str,
-    hue: str,
-    output_filename: str | None = None,
-) -> None:
-    graph = sns.relplot(
-        data=data,
-        x=x_axis,
-        y=y_axis,
-        col=hue,
-        hue=hue,
-        facet_kws=dict(sharex=True, sharey=True),
-        col_wrap=3,
+    save_plot_as_png(
+        graph,
+        get_output_path(data, sub_dir, plot_name),
     )
 
-    x_axis_label, y_axis_label = get_axis_labels(x_axis, y_axis, group)
-    graph.set_axis_labels(x_axis_label, y_axis_label)
-
-    if output_filename is not None:
-        save_plot_as_png(
-            graph,
-            Path(__file__).parents[2]
-            / "data"
-            / "plots"
-            / f"{group}_subplot_relplot_{output_filename}.png",
-        )
 
+def get_axis_labels(
+    benchmark_df: pd.DataFrame, x_axis: str, y_axis: str
+) -> tuple[str, str]:
+    group = benchmark_df.group.unique()
+    if len(group) != 1:
+        raise ValueError("Expected only one group value in dataframe")
 
-def get_axis_labels(x_axis: str, y_axis: str, group: str) -> tuple[str, str]:
     if x_axis.startswith("stats"):
-        x_axis_label = f"{x_axis.split('.')[-1]} {group} time (s)"
+        x_axis_label = f"{x_axis.split('.')[-1]} {group[0]} time (s)"
     else:
         x_axis_label = x_axis.capitalize().replace("_", " ")
 
     if y_axis.startswith("stats"):
-        y_axis_label = f"{y_axis.split('.')[-1]} {group} time (s)"
+        y_axis_label = f"{y_axis.split('.')[-1]} {group[0]} time (s)"
     else:
         y_axis_label = y_axis.capitalize().replace("_", " ")
 
     return x_axis_label, y_axis_label
 
 
+def get_output_path(benchmarks_df: pd.DataFrame, sub_dir: str, plot_name: str) -> Path:
```

Comment:
```suggestion
def get_output_path(benchmarks_df: pd.DataFrame, sub_dir: Path, plot_name: str) -> Path:
```
Seems like a directory should have `Path` type?